cmd/mount.go

```diff
@@ -382,6 +394,51 @@ func sigusrHandler(pipeline *internal.Pipeline, ctx context.Context) daemon.Sign
 	}
 }
 
+func setGOConfig() {
+	// Ensure we always have more than 1 OS thread running goroutines, since there are issues with having just 1.
```

Comment:
why do we need more than 1? is this so that the profiler runs on a separate thread?

Reply:
We don't need more then 1. If you see the earlier implementation, it runs only when you run blobfuse2 in foreground (which is purely for debugging and not for production). This profiler will run for both background and foreground and it runs as a separate thread. Earlier profiler dumps report only on successful completion and bring down of binary (i.e., unmount in our case), while this is more dynamic and on a running system anytime you wish to take a snapshot of profiling info, you can do it through browser or using "go tool pprof" command. So technically both are different things, meant for different purpose. When in our regular use-case only this will be available (as other one runs only in foreground mode).